lib/src/features/account/widget/button_change_avatar_portals.dart

```diff
@@ -0,0 +1,126 @@
+        child: OverlayPortal(
+          controller: _controller,
+          overlayChildBuilder: (BuildContext context) {
+            return Positioned(
+              top: 5,
+              right: 5,
+              child: Container(
+                height: 300,
+                width: 200,
+                decoration: BoxDecoration(
+                  borderRadius: BorderRadius.circular(20),
+                  color: AppColors.lightGray,
+                ),
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    _buildButtonPortals(
+                      title: S.of(context).from_gallery,
+                      onPressed: () {
+                        _pickImageFromGallery(context);
+                      },
+                    ),
+                    _buildButtonPortals(
+                      title: S.of(context).from_camera,
+                      onPressed: () {
+                        _pickImageFromCamera(context);
+                      },
+                    ),
+                  ],
+                ),
+              ),
+            );
+          },
+          child: Text(
+            S.of(context).change_avatar,
+            style: AppStyles.blackTextSmall,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildButtonPortals({
+    required String title,
+    required onPressed,
+  }) {
+    return XTextButton(
+      title: title,
+      style: AppStyles.blackTextLarge,
+      onPressed: onPressed,
+    );
+  }
+
+  Future _pickImageFromGallery(BuildContext context) async {
+    final PermissionStatus permissionStatus = await Permission.photos.request();
+    if (permissionStatus == PermissionStatus.granted) {
+      final image = await ImagePicker().pickImage(source: ImageSource.gallery);
+      final String fileName = StringUtils.generateIdImageName();
+      if (image != null) {
+        await ref.add(
+          folder: StorageFolder.users,
+          type: StorageType.image,
+          file: File(image.path),
+          fileName: fileName,
+        );
+        if (context.mounted) {
+          await context.read<AccountBloc>().onUpdateAvatar(fileName);
+        }
+      }
+    }
+  }
+
+  Future _pickImageFromCamera(BuildContext context) async {
+    final PermissionStatus permissionStatus = await Permission.camera.request();
+    if (permissionStatus == PermissionStatus.granted) {
+      final image = await ImagePicker().pickImage(source: ImageSource.camera);
+      final String fileName = StringUtils.generateIdImageName();
+      if (image != null) {
+        await ref.add(
+          folder: StorageFolder.users,
+          type: StorageType.image,
+          file: File(image.path),
+          fileName: fileName,
+        );
+        if (context.mounted) {
+          await context.read<AccountBloc>().onUpdateAvatar(fileName);
+        }
```

Comment:
Move a file `picker_utils.dart` to reuse in future
And separate permission to `permission_utils.dart` and handle show dialogue if the user doesn't grant